### PR TITLE
(fix) gate_io native token as fee

### PR DESCRIPTION
**Pull request by MementoRC**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
https://snapshot.org/#/hbot-prp.eth/proposal/QmR21ZyXJ7a4VqUC1SX3ensZ747UaymMn57mV59zQw4YEr
Providing a workaround to the issue:
https://github.com/hummingbot/hummingbot/issues/5129
Since GT is not on Binance, the fee GT-USDT price is not found and that generates a warning. The fee is thus not converted into USDT (not sure it would be though)

**Tests performed by the developer**:
Ran the updated bot and waited for a trade, then ran history ( no change to the conf_global.yml or conf_fee_override.yml)
image

**Tips for QA testing**:

### CODE REVIEW

**Comment by MementoRC** [comment]
PRP: https://snapshot.org/#/hbot-prp.eth/proposal/QmR21ZyXJ7a4VqUC1SX3ensZ747UaymMn57mV59zQw4YEr

**Comment by MementoRC** [comment]
I will try to update the test suite

File: hummingbot/connector/exchange/gate_io/gate_io_exchange.py [hunk 1]
```diff
@@ -420,14 +420,17 @@ def _process_trade_message(self, trade: Dict[str, Any], client_order_id: Optiona
             self.logger().debug(f"Ignoring trade message with id {client_order_id}: not in in_flight_orders.")
             return
 
+        # When GT token is used to pay fee 'gt_fee' is provided by the API
+        if 'gt_fee' in trade and Decimal(trade["gt_fee"]) > 0:
+            flat_fees = [TokenAmount(amount=Decimal(trade["gt_fee"]), token='GT')]
+        else:
+            flat_fees = [TokenAmount(amount=Decimal(trade["fee"]), token=trade["fee_currency"])]
+
```
Comment (aarmoa) [inline hummingbot/connector/exchange/gate_io/gate_io_exchange.py:428]: The API documentation does not discard the case where fee is paid both in GT and in another token. We should prepare the logic for that scenario, by adding a flat fee for GT fee (if there is any GT fee) and a flat fee for other token fee (if there is and it is not 0).

-- Reviewed in representation of Hummingbot Foundation Technical DAO

**Review by aarmoa — CHANGES_REQUESTED** [review]

File: hummingbot/connector/exchange/gate_io/gate_io_exchange.py (hunk 1 above)
Comment (MementoRC) [reply hummingbot/connector/exchange/gate_io/gate_io_exchange.py:428]: Gate.io support had replied to my ticket related to this, that if I set the 'Pay with GT' flag, I cannot use my points, thus I did not think about that. Thanks for pointing that out, in Gate.io case, this also requires to check and add the 'point_fee'
Now the display will be tricky as fee is only displayed in 1 currency or should this feature be also supported by the 'history' output?

Comment (cardosofede) [reply hummingbot/connector/exchange/gate_io/gate_io_exchange.py]: Hi @aarmoa @MementoRC,

Today we discuss this issue in the Hummingbot Foundation Meeting and we have a proposal for this situation, but we would like to let the decision open to the community.
We think that the fees has to admit payments in the native token of the exchanges (like GT in this case) or base/quote, but POINTS are only used in Gate.io so we think we don't need to include it.

**Comment by MementoRC** [comment]
Should this remain open open?

**Comment by MementoRC** [comment]
Actually, this seems obsolete

**Pull request closed without merging**